### PR TITLE
Fix issue #742

### PR DESCRIPTION
**Pull request by tr3ee**
Fix ParseZone() parsing error #742 
 - fix error comparison in setTA() in scan_rr.go
 - add testcase to make sure everything is ok

### CODE REVIEW

File: scan_test.go [hunk 1]
```diff
@@ -79,3 +79,20 @@ func TestParseZoneInclude(t *testing.T) {
 		}
 	}
 }
+
+func TestParseZoneTA(t *testing.T) {
+	r := strings.NewReader(` Ta 0 0 0`)
```
Comment (miekg) [inline scan_test.go]: This should also break with NewRR() which leads to a shorter testcase

Comment (tr3ee) [reply scan_test.go]:
```Go
func TestParseZoneTA(t *testing.T) {
	rr, err := dns.NewRR(` Ta 0 0 0`)
	if err != nil {
		t.Fatalf("expected no error, but got %s", err)
	}
	if rr == nil {
		t.Fatalf(`expected a normal RR, but got nil`)
	}
}
```
what about this then?

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [miekg/dns] Fix issue #742 (#74..." ]
>tr3ee commented on this pull request.
>
>
>
>> @@ -79,3 +79,20 @@ func TestParseZoneInclude(t *testing.T) {
> 		}
> 	}
> }
>+
>+func TestParseZoneTA(t *testing.T) {
>+	r := strings.NewReader(` Ta 0 0 0`)
>
>```Go
>func TestParseZoneTA(t *testing.T) {
>	rr, err := dns.NewRR(` Ta 0 0 0`)
>	if err != nil {
>		t.Fatalf("expected no error, but got %s", err)
>	}
>	if rr == nil {
>		t.Fatalf(`expected a normal RR, but got nil`)
>	}
>}
>```
>what about this then?

yep. Except the name can just be TestParseTA and the second Fatalf can just be 
Fatal

**Comment by tr3ee** [comment]
> [ Quoting <notifications@github.com> in "Re: [miekg/dns] Fix issue #742 (#74..." ]
> tr3ee commented on this pull request. > @@ -79,3 +79,20 @@ func TestParseZoneInclude(t *testing.T) { } } } + +func TestParseZoneTA(t *testing.T) { +	r := strings.NewReader(` Ta 0 0 0`) ```Go func TestParseZoneTA(t *testing.T) { rr, err := dns.NewRR(` Ta 0 0 0`) if err != nil { t.Fatalf("expected no error, but got %s", err) } if rr == nil { t.Fatalf(`expected a normal RR, but got nil`) } } ``` what about this then?
> yep. Except the name can just be TestParseTA and the second Fatalf can just be Fatal

updated : )

**Pull request merged**